text/0000-nested-publish.md

```diff
@@ -119,9 +119,10 @@ Then you can `cargo publish` from within the parent directory `foo/`, and this w
 
 Two new possible values are added to the manifest.
 
-* The `package.publish` field allows `"nested"` as a value, in addition to existing `false` and `true`. This value affects `cargo publish` and nested dependencies as described below.
+* The `package.publish` or `workspace.package.publish` field allows `"nested"` as a value, in addition to existing `false` and `true`. This value affects `cargo publish` and nested dependencies as described below.
 * The `dependencies.*.publish` field is newly defined, with the only currently allowed value being `"nested"`, to declare that that dependency is a nested dependency. (If desired, `publish = false` could be also be permitted to explicitly document an intent not to nest; this would be identical to the status quo and to omitting the key.)
     * It is an error if a nested dependency does not have a `path` field, or if it has a `version`, `git`, or any other package source field, unless future work defines a meaning for that combination.
+    * Workspace inheritance is not permitted; `workspace.dependencies.*.publish` is an error at `cargo package`/`cargo publish` time. (Builds should ignore the field, for forward compatibility.)
```

Comment:
Why is this only an error on publish and not on build / parse?  We add new fields all the time and break people.  Its part of why the unused manifest key warning exists, to let people know they are in danger of being broken.

Reply:
I think what I was thinking was that if inherited `workspace.dependencies.*.publish` becomes a thing, it shouldn't break builds by users who just want to build the code and aren't responsible for publishing it. But we can certainly start with making it unconditionally an error.